include/picongpu/particles/boundary/Absorbing.hpp

```diff
@@ -83,6 +89,157 @@ namespace picongpu
+                 * This means, using the paper notation, for us the integral is not from t_i to t but each time only
+                 * over the current time step.
+                 * As the particle's weight already has a combined effect from t_i to previous time step.
+                 * If a particle never leaves PML after entering it, these formulations are equivalent.
+                 *
+                 * @TODO add an optional extra particle attribute to store additional weighting if a user chose that -
+                 * then particles leaving PML would have their weights "restored", unlike in the current version.
+                 */
+                struct DampWeightsInPml : public functor::misc::Parametrized<DampWeightsInPmlParameters>
+                {
+                    //! Name, required to be wrapped later
+                    static constexpr char const* name = "dampWeightsInPml";
+
+                    /** Process the current particle located in the given cell
+                     *
+                     * @param offsetToTotalOrigin offset of particle cell in the total domain
+                     * @param particle handle of particle to process (can be used to change attribute values)
+                     */
+                    template<typename T_Particle>
+                    DINLINE void operator()(DataSpace<simDim> const& offsetToTotalOrigin, T_Particle& particle)
+                    {
+                        auto const axis = m_parameters.axis;
+                        // Velocity half time step back relative to the current position
+                        auto velocity = Velocity{};
+                        auto const vel
+                            = velocity(particle[momentum_], attribute::getMass(particle[weighting_], particle))[axis];
+
+                        // Positions in local domain, in units of cells, without guards
+                        auto const localCellIdx = offsetToTotalOrigin - m_parameters.localToTotalDomainOffset;
+                        auto const positionCurrent
+                            = (precisionCast<float_X>(localCellIdx) + particle[position_])[axis];
+                        auto const positionPrevious = positionCurrent - vel * DELTA_T / cellSize[axis];
+
+                        // Integral over the last time step as described in comment of this struct
+                        auto const timeStepIntegral = fields::absorber::pml::getNormalizedSigmaIntegral(
+                            positionPrevious,
+                            positionCurrent,
+                            DELTA_T,
+                            m_parameters.localParameters,
+                            axis);
+
+                        // Adjust weighting according to [Lehe2022], cap at MIN_WEIGHTING
+                        auto const weightingMultiplier = math::exp(-timeStepIntegral);
+                        auto const dampedWeighting = float_64(particle[weighting_]) * weightingMultiplier;
```

Comment:
is 64bit required ?

Reply:
I was not sure here and it is not performance-critical

Reply:
I checked - it is fine as `float_X`, changed to that.